### PR TITLE
2823 add missing filters events

### LINKED ISSUE #2823: 🐛Events - Past events not appearing in filter list
@wicksipedia 

### Describe the bug
The events page's counts do not include the aggregated past events in the filters until they've been loaded client side.

### Note
The fix probably involves doing a separate fetch to return the total number of events and using that number to populate the events filters.

### Tasks
- [ ] Investigate and fix it

### Screenshots

Image

**Figure**: **Note how the number of .Net Core events increases after more events load**

### PR DESCRIPTION
**Pull request by Calinator444**
### Affected Routes
- ssw.com.au/events

### Relevant issue
- Fixed #2823 

### Changes
- 🐛 Event filters will now show filters for the full list of events
- ✨ Clicking on a filter will load all events under that category
- ⚡ Once an event filter has been applied, events under that filter will be cached next time that filter is applied
- ♻️ Removed nested ternary operator used to handle loading state for events
- 🚸 Filtering for "Angular and React" events will now display Angular events
- 🧹 Tidied up events that were missing categories/had duplicate categories with different casing

### Screenshots

fetching functionality for events

**Figure**: **New fetching functionality for events that haven't been loaded**

<br>

event caching

**Figure**: **Caching behavior for filters that were previously applied**

### CODE REVIEW

File: components/filter/events.tsx [hunk 1]
```diff
@@ -47,31 +51,42 @@ export type EventTrimmed = {
...
+  filterCategories,
   sidebarBody,
   defaultToPastTab,
 }: EventsFilterProps) => {
   const [pastSelected, setPastSelected] = useState<boolean>(defaultToPastTab);
+  const { past, upcoming } = filterCategories;
+  const { filters: futureFilters } = useEvents(upcoming);
+  const { filters: pastFilters } = useEvents(past);
+
+  useEffect(() => {}, [pastFilters, futureFilters]);
```
Comment (amankumarrr) [inline components/filter/events.tsx]: do we need it there?

File: components/filter/events.tsx [hunk 2]
```diff
@@ -140,87 +150,101 @@ const EventTab = ({ children }: { children: React.ReactNode }) => {
...
   We need this because there's an issue preventing us from syncing the files in the repo
   to Tina cloud. Images that aren't synced will 404.
    
    */
-  const [thumbnail, setFallbackImage] = useState(event.thumbnail);
+  const [oldTitle, setOldTitle] = useState(event.title);
+  const [visible, setVisible] = useState(false);
+  const [thumbnail, setFallbackImage] = useState("");
+  useEffect(() => {
+    if (event.title !== oldTitle) setVisible(false);
```
Comment (amankumarrr) [inline components/filter/events.tsx]: can we put parenthesis around here? as per this rule - https://www.ssw.com.au/rules/how-to-avoid-problems-in-if-statements/#:~:text=See%20history.%20Try%20to

File: components/filter/events.tsx [hunk 3]
```diff
@@ -140,87 +150,101 @@ const EventTab = ({ children }: { children: React.ReactNode }) => {
...
+    setTimeout(() => {
+      setVisible(true);
+      setOldTitle(event.title);
+    }, 100);
```
Comment (amankumarrr) [inline components/filter/events.tsx]: can we avoid using timeout here?

**Review by amankumarrr — CHANGES_REQUESTED** [review]
Looks good to me, just left a few comments, let me know if it doesn't make sense.

File: components/filter/events.tsx (hunk 1 above)
Comment (Calinator444) [reply components/filter/events.tsx]: Oh, my bad. I've deleted that now.

File: components/filter/events.tsx (hunk 2 above)
Comment (Calinator444) [reply components/filter/events.tsx]: Done - I've added parentheses around the if statement

File: components/filter/events.tsx (hunk 3 above)
Comment (Calinator444) [reply components/filter/events.tsx]: As per our conversation, I've improved the logic for hiding events so we no longer need a timeout

**Review by amankumarrr — APPROVED** [review]

**Pull request merged**